doc/development_process/api_lifecycle.rst

```diff
@@ -198,7 +198,10 @@ The following are the requirements for deprecating an existing API:
   where an API is deprecated sooner.
 - What is required when deprecating:
 
-  - Mark as deprecated
+  - Mark as deprecated. This can either be done using the ``__deprecated``
+    compiler directive or by introducing a Kconfig option (typically one that
```

Comment:
Propose:

> This can be done by using the `__deprecated` qualifier on a function declaration (`__DEPRECATED_MACRO` marker in a macro definition), or by introducing....

`__deprecated` is implemented using a function attribute which is a compiler directive, but syntactically it looks more like a qualifier.  Also mentioning the way to deprecate macros seems worthwhile.

Reply:
Thanks for the suggestion @pabigot, just incorporated it.